Sources/AWSLambdaRuntimeCore/LambdaHandler.swift

```diff
@@ -26,13 +25,16 @@ import NIOCore
 ///         level protocols ``EventLoopLambdaHandler`` and
 ///         ``ByteBufferLambdaHandler``.
 @available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
-public protocol LambdaHandler: EventLoopLambdaHandler {
-    /// The Lambda initialization method.
-    /// Use this method to initialize resources that will be used in every request.
-    ///
-    /// Examples for this can be HTTP or database clients.
-    /// - parameters:
-    ///     - context: Runtime ``LambdaInitializationContext``.
+public protocol LambdaHandler {
+    /// The lambda functions input. In most cases this should be `Codable`. If your event originates from an
+    /// AWS service, have a look at [AWSLambdaEvents](https://github.com/swift-server/swift-aws-lambda-events),
+    /// which provides a number of commonly used AWS Event implementations.
+    associatedtype Event
+    /// The lambda functions output. Can be `Void`.
+    associatedtype Output
+
+    init()
+
     init(context: LambdaInitializationContext) async throws
```

Comment:
I still think we need two different handler protocols to enable this easy use. See below re infinite recursion.

Reply:
we could, I was debating this myself. I was also expecting infinite recursion but strangely it was not the case